src/nvim/option.c

```diff
@@ -3336,8 +3336,8 @@ void set_tty_background(const char *value)
   if (starting) {
     // Wait until after startup, so OptionSet is triggered.
     do_cmdline_cmd((value[0] == 'l')
-                   ? "autocmd VimEnter * ++once ++nested set bg=light"
-                   : "autocmd VimEnter * ++once ++nested set bg=dark");
+                   ? "autocmd VimEnter * ++once ++nested if &bg == '' | set bg=light | endif"
```

Comment:
I don't think this would work. need `!nvim_get_option_info2("bg", {}).was_set`

Reply:
I personally can't reproduce the original issue

Do you know how to write a test for this racing condition?